src/logger.js

```diff
@@ -44,7 +44,14 @@ export type LoggerType = {|
 |};
 
 function httpTransport({ url, method, headers, json } : {| url : string, method : string, headers : { [string] : string }, json : Object |}) : ZalgoPromise<void> {
-    return request({ url, method, headers, json }).then(noop);
+    if (window.navigator.sendBeacon) {
+        return new ZalgoPromise(resolve => {
+            const blob = new Blob([ JSON.stringify(json) ], headers);
```

Comment:
The sendBeacon() function has limited support for headers. It seems like it will only pass through the content type. 

I did a quick test locally to test out the behavior and `sendBeacon()` does not pass the 'Test-Key' header in the snippet below:

```js
  const url = '/logger';
  const data = {
    fun: true,
    js: true
  };
  const headers = {
    type: 'application/json',
    'Test-Key': 'Test-Value'
  };

  const blob = new Blob([ JSON.stringify(data) ], headers);
  // does not send the 'Test-Key' header
  window.navigator.sendBeacon(url, blob);

  // sends both header values
  fetch(url, {
    method: 'POST',
    body: JSON.stringify(data),
    keepalive:true,
    headers
  });
``` 

The lack of header support leads me to believe this would be a breaking change for the `httpTransport()` function.

Reply:
Yeah, you’re absolutely right.  I have been meaning to remove that and have been working on something else. I should have fixed this to not waste your time. Inspected the headers being sent from the logger and we are good there.  The size is capped to about 64kb and we are good on that limitation as well.

Reply:
I did a quick scan of our SDK and looks like nobody is actually passing in custom headers. But for backwards compat, can we just say if any custom headers are passed, fall back to `request`?

Reply:
@bluepnume Sounds good!  I am going to remove the usage of a blob and just send along the json.  I will also check for any headers being passed and fallback if that is the case.